lib/response-handler.js

```diff
@@ -74,19 +74,20 @@ module.exports = (bluebird, error, logger, parseLinkHeader, URI) => {
      *
      * This can be used for both link headers and location headers.
      *
-     * @param {string} url
+     * @param {string} resourceUrl
      * @param {string} link
      * @return {string}
      */
-    function resolveLink(url, link) {
-        var uri;
+    function resolveLink(resourceUrl, link) {
+        var resolvedLink, uri;
 
         logger.debug("In responseHandler.resolveLink()");
-        logger.debug(`Url param: ${url}. Link param: ${link}`);
+        logger.debug(`Url param: ${resourceUrl}. Link param: ${link}`);
 
-        uri = new URI(link, url);
+        uri = new URI(link, resourceUrl);
+        resolvedLink = uri.toString();
```

Comment:
Do this only on the path. What if my uri was... `https://shelf-api.artifact.com/blah/artifact/...`